arm11/source/start.s

```diff
@@ -58,6 +58,12 @@ _start:
 	@ Clear exclusive records
 	clrex
 
+	@ Get the CPUID
+	@ Branch to secondary boot if its not CPU0
+	mrc p15, 0, r0, c0, c0, 5
+	ands r0, r0, #3
```

Comment:
prefer `tst r0, #3` to `ands r0, r0, #3`?

Reply:
no, r0 can have some upper bits be non-zero so it always needs to mask those out and keep only the two low bits. there are more details in the ARM11 TRM, linked somewhere in the wiki

Reply:
ah right we use `r0` later.